src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
```

Comment:
The class `MachineServiceImpl` is implementing `MachineService<Machine>`, which means it is not fully utilizing the generic type `T`. Consider parameterizing `MachineServiceImpl` with a specific type to make it more flexible.